src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "../api/api";
+import { getTimeStr } from "../utils/getTimeStr";
+
+function SingleArticle() {
+  const { article_id } = useParams();
+  const [currentArticle, setCurrentArticle] = useState();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getArticleById(article_id).then((article) => {
+      console.log(article);
+      setCurrentArticle(article);
+
+      setLoading(false);
+    });
+  }, [article_id]);
+
+  if (!loading) {
```

Comment:
You would normally have an else here to display visual information to the user that the page is still loading